### PR TITLE
[POAE7-2380] [HotFix] Groupby HashTable Code Refactor and Bugfix

### PR DESCRIPTION
**Pull request by bigPYJ1151**
### What changes were proposed in this pull request?

- Remove keyless and columnar-layout related code in CiderAggHashTable.
- Fix offset bug under group-by w/o aggregations.
- Fix mistake introduced by previous conflict resolve.

### Why are the changes needed?

Bug fix.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs

### Which label does this PR belong to?

BugFix

Pls take a review @harborn @winningsix

### CODE REVIEW

File: cider/exec/operator/aggregate/CiderAggHashTable.cpp [hunk 1]
```diff
@@ -312,7 +307,7 @@ std::vector<CiderAggHashTableEntryInfo> CiderAggHashTable::fillColsInfo() {
   std::vector<CiderAggHashTableEntryInfo> cols_info(columns_num_);
 
   size_t offset = 0, col_slot_context_index = 0;
-  bool is_key = !is_keyless_;
+  bool is_key = true;
```
Comment (Deegue) [inline cider/exec/operator/aggregate/CiderAggHashTable.cpp:310]: Do we always have one or more keys here no matter it is in count or avg?

**Review by Deegue — COMMENTED** [review]
LGTM. Just one question above, and I'm not sure why we should remove is_keyless. Maybe it's also related to Arrow format?

**Review by harborn — APPROVED** [review]

**Comment by bigPYJ1151** [comment]
> LGTM. Just some questions, and I'm not sure why we should remove is_keyless. Maybe it's also related to Arrow format?

keyless hash is a trivial optimization which is not in use, so just remove it.

File: cider/exec/operator/aggregate/CiderAggHashTable.cpp (hunk 1 above)
Comment (bigPYJ1151) [reply cider/exec/operator/aggregate/CiderAggHashTable.cpp:310]: For group-by there will be one key col at least.

**Review by winningsix — APPROVED** [review]

**Pull request merged**